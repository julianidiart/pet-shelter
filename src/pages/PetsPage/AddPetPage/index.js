import React, { Component } from "react";
import { connect } from "react-redux";
import PetForm from "../../../components/PetForm";
import { startAddPet } from "../../../actions/pets";

export class AddPetPage extends Component {
  onSubmit = pet => {
    this.props.startAddPet(pet);
    this.props.history.push("/pets");
  };
  render() {
    return (
      <div>
        <PetForm onSubmit={this.onSubmit} />
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  startAddPet: pet => dispatch(startAddPet(pet))
});

export default connect(
  undefined,
  mapDispatchToProps
)(AddPetPage);
