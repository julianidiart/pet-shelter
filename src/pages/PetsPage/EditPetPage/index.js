import React, { Component } from "react";
import { connect } from "react-redux";
import PetForm from "../../../components/PetForm";
import { editPet, removePet } from "../../../actions/pets";

export class EditPetPage extends Component {
  onSubmit = pet => {
    this.props.editPet(this.props.pet.id, pet);
    this.props.history.push("/pets");
  };
  onClick = () => {
    this.props.removePet(this.props.pet.id);
    this.props.history.push("/pets");
  };
  render() {
    return (
      <div>
        <PetForm pet={this.props.pet} onSubmit={this.onSubmit} />
        <button onClick={this.onClick}>Remove</button>
      </div>
    );
  }
}

const mapStateToProps = (state, props) => {
  return {
    pet: state.pets.find(pet => pet.id === props.match.params.id)
  };
};

const mapDispatchToProps = dispatch => ({
  editPet: (id, pet) => dispatch(editPet(id, pet)),
  removePet: id => dispatch(removePet(id))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(EditPetPage);
