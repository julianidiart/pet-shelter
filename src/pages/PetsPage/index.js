import React from "react";

import PetsList from "../../components/PetsList";

export default props => (
  <div>
    <button onClick={() => props.history.push("/add-pet")}>Add pet</button>
    <PetsList />
  </div>
);
