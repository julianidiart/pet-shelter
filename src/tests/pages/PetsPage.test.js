import React from "react";
import { shallow } from "enzyme";
import PetsPage from "../../pages/PetsPage";

test("should render PetsPage correctly", () => {
  const wrapper = shallow(<PetsPage />);
  expect(wrapper).toMatchSnapshot();
});
