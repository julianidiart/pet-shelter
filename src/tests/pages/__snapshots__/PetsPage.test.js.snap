// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render PetsPage correctly 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    Add pet
  </button>
  <Connect(PetsList) />
</div>
`;
